src/main/kotlin/com/sourcegraph/cody/agent/CodyAgent.kt

```diff
@@ -102,9 +102,9 @@ class CodyAgent(private val project: Project) : Disposable {
     }
   }
 
-  fun shutdown() {
-    val server = getServer(project) ?: return
-    executorService.submit<CompletableFuture<Void>> {
+  fun shutdown(): Future<out CompletableFuture<Void>?>? {
```

Comment:
`server.shutdown` returns `CompletableFuture`, what is the reason to additionally run it on a separate thread using `executorService`?
Wouldn't that be sufficient?
 
```kotlin
  fun shutdown(): CompletableFuture<Void>? {
    val server = getServer(project)
    return server?.shutdown()?.thenAccept {
      server.exit()
      agentNotRunningExplanation = "Cody Agent shut down"
      listeningToJsonRpc.cancel(true)
    }
  }
```